packages/react/src/primitives/Input/Input.tsx

```diff
@@ -33,40 +34,47 @@ const InputPrimitive: Primitive<InputProps, 'input'> = (
     ...rest
   },
   ref
-) => (
-  <View
-    aria-invalid={hasError}
-    as="input"
-    autoComplete={autoComplete}
-    checked={checked}
-    className={classNames(
-      ComponentClassNames.Input,
-      ComponentClassNames.FieldGroupControl,
-      className
-    )}
-    data-size={size}
-    data-variation={variation}
-    defaultChecked={defaultChecked}
-    defaultValue={defaultValue}
-    isDisabled={isDisabled}
-    id={id}
-    onBlur={onBlur}
-    onChange={onChange}
-    onCopy={onCopy}
-    onCut={onCut}
-    onFocus={onFocus}
-    onInput={onInput}
-    onPaste={onPaste}
-    onSelect={onSelect}
-    onWheel={onWheel}
-    readOnly={isReadOnly}
-    ref={ref}
-    required={isRequired}
-    type={type}
-    value={value}
-    {...rest}
-  />
-);
+) => {
+  const componentClasses = classNames(
+    ComponentClassNames.Input,
+    ComponentClassNames.FieldGroupControl,
+    classNameModifier(ComponentClassNames.Input, variation),
```

Comment:
Need to update unit tests to reflect the new classnames being added to each primitive